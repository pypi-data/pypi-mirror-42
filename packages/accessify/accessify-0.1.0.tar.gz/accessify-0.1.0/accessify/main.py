"""
Provide implementation of accessibility levels.
"""
import copy
import inspect

from accessify.errors import (
    INACCESSIBLE_DUE_TO_ITS_PROTECTION_LEVEL_EXCEPTION_MESSAGE,
    InaccessibleDueToItsProtectionLevelException,
)

ACCESS_WRAPPERS_NAMES = (
    'private_wrapper',
    'protected_wrapper',
)


def accessify(cls):
    """
    Mark class as class that uses accessibility levels.

    Rules:
        - check if called method is covered by private access decorator,
        - remove method that covered by private access decorator from __dir__.
    """
    class_locals = copy.deepcopy(dir(cls))

    for name, func in list(cls.__dict__.items()):

        if hasattr(func, '__name__'):

            if func.__name__ in ACCESS_WRAPPERS_NAMES:
                class_locals.remove(name)

    def dir_magic_method_mock(_):
        return class_locals

    cls.__dir__ = dir_magic_method_mock

    return cls


def does_parents_contain_private_method(classes, method):
    """
    Check if at least one of parent classes contain method.

    Rules:
        - if classes first element is object, so obviously return false,
        - if class parents contain the method (called by child) and this method has private access level, return true.
    """
    if classes[0].__name__ == 'object':
        return False, None

    for class_ in classes:
        if hasattr(class_, method.__name__):
            if getattr(class_, method.__name__).__name__ in ACCESS_WRAPPERS_NAMES:
                return True, class_

    return False, None


def find_decorated_method(function):
    """
    Return the method going from the most top placed decorator under it.
    """
    if function.__class__.__name__ in ('staticmethod', 'classmethod'):
        return find_decorated_method(function.__func__)

    if function.__closure__ is not None:
        return find_decorated_method(function.__closure__[0].cell_contents)

    return function


def private(func):
    """
    Provide private accessibility level.
    """
    def private_wrapper(*args, **kwargs):
        """
        Provide private accessibility level wrapper.
        """
        instance, *arguments_without_instance = args
        instance_class = instance.__class__
        instance_class_parents = instance.__class__.__bases__

        method = find_decorated_method(function=func)

        does_contain, class_contain = does_parents_contain_private_method(classes=instance_class_parents, method=method)

        if class_contain:
            raise InaccessibleDueToItsProtectionLevelException(
                INACCESSIBLE_DUE_TO_ITS_PROTECTION_LEVEL_EXCEPTION_MESSAGE.format(
                    class_name=class_contain.__name__, method_name=method.__name__,
                )
            )

        if inspect.currentframe().f_back.f_locals.get('self') is None:
            raise InaccessibleDueToItsProtectionLevelException(
                INACCESSIBLE_DUE_TO_ITS_PROTECTION_LEVEL_EXCEPTION_MESSAGE.format(
                    class_name=instance_class.__name__, method_name=method.__name__,
                )
            )

        if func.__class__.__name__ == 'classmethod':
            arguments = (instance_class, ) + tuple(arguments_without_instance)
            return func.__func__(*arguments, **kwargs)

        elif func.__class__.__name__ == 'staticmethod':
            return func.__func__(*arguments_without_instance, **kwargs)

        else:
            return func(*args, **kwargs)

    return private_wrapper


def protected(func):
    """
    Provide private accessibility level.
    """
    def protected_wrapper(*args, **kwargs):
        """
        Provide private accessibility level wrapper.
        """
        instance, *arguments_without_instance = args
        instance_class = instance.__class__

        method = find_decorated_method(function=func)

        if inspect.currentframe().f_back.f_locals.get('self') is None:
            raise InaccessibleDueToItsProtectionLevelException(
                INACCESSIBLE_DUE_TO_ITS_PROTECTION_LEVEL_EXCEPTION_MESSAGE.format(
                    class_name=instance_class.__name__, method_name=method.__name__,
                )
            )

        instance, *arguments_without_instance = args
        instance_class = instance.__class__

        if func.__class__.__name__ == 'classmethod':
            arguments = (instance_class, ) + tuple(arguments_without_instance)
            return func.__func__(*arguments, **kwargs)

        elif func.__class__.__name__ == 'staticmethod':
            return func.__func__(*arguments_without_instance, **kwargs)

        else:
            return func(*args, **kwargs)

    return protected_wrapper
