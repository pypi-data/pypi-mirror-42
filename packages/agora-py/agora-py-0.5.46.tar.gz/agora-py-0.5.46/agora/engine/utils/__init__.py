"""
#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=#
  Ontology Engineering Group
        http://www.oeg-upm.net/
#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=#
  Copyright (C) 2016 Ontology Engineering Group.
#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=#
  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

            http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=#
"""
import os
from threading import Lock

__author__ = 'Fernando Serena'


class Singleton(type):
    _lock = Lock()

    def __call__(cls, *args, **kwargs):
        with cls._lock:
            id = kwargs.get('id', '')
            if id not in cls.instances:
                cls.instances[id] = super(Singleton, cls).__call__(*args, **kwargs)
            return cls.instances[id]


def prepare_store_path(base, path):
    if base and not os.path.exists(base):
        os.makedirs(base)
    if base or path:
        full_path = '/'.join(filter(lambda x: x, [base, path]))
        if not os.path.exists(full_path):
            os.makedirs(full_path)
        return full_path


class Semaphore(object):
    def __init__(self):
        self.value = -1

    def __enter__(self):
        self.value = 0

    def __exit__(self, exc_type, exc_val, exc_tb):
        self.value = 1

    def isSet(self):
        return self.value == 1

    def set(self):
        self.value = 1


def get_immediate_subdirectories(a_dir):
    if not os.path.exists(a_dir):
        os.makedirs(a_dir)
    return [name for name in os.listdir(a_dir)
            if os.path.isdir(os.path.join(a_dir, name))]


class Wrapper(object):
    def __init__(self, object):
        self.__lock = Lock()
        self.__attr_locks = {}

        if isinstance(object, Wrapper):
            self.__object = object.__object
            self.__argret = object.__argret
            self.__attr_locks = object.__attr_locks
            self.__lock = object.__lock
        else:
            self.__object = object
            self.__argret = {}

    def decorate(self, f):
        def wrapper(*args, **kwargs):

            key = f.func_name + str(args)
            with self.__lock:
                if key not in self.__attr_locks:
                    self.__attr_locks[key] = Lock()

            with self.__attr_locks[key]:
                if key not in self.__argret:
                    ret = f(*args, **kwargs)

                    if kwargs:
                        return ret
                    if ret is not None:
                        self.__argret[key] = ret

                return self.__argret.get(key, None)

        return wrapper

    def __getattr__(self, item):

        if item.startswith('__'):
            return self.__object.__getattribute__(item)

        with self.__lock:
            if item not in self.__attr_locks:
                self.__attr_locks[item] = Lock()

        with self.__attr_locks[item]:
            if item not in self.__argret:
                attr = self.__object.__getattribute__(item)
                if hasattr(attr, '__call__'):
                    return self.decorate(attr)
                else:
                    self.__argret[item] = attr

        return self.__argret[item]


stopped = Semaphore()
