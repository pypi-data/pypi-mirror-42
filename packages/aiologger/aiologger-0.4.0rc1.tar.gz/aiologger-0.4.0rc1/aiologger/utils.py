from typing import Callable


class classproperty:
    def __init__(self, func):
        self._func = func

    def __get__(self, obj, owner):
        return self._func(owner)


class CallableWrapper:
    def __init__(self, func: Callable) -> None:
        self.func = func

    def __call__(self, *args, **kwargs):
        return self.func(*args, **kwargs)
