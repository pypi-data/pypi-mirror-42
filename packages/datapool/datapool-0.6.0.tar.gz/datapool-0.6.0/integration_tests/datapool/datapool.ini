[db]
connection_string = sqlite+pysqlite:////Users/uweschmitt/Projects/eawag_datapool/src/integration_tests/lz/.simple.db

[backup_landing_zone]
folder = 

[r]
extension = .r
executable = /usr/local/bin/R

[matlab]
extension = .m
executable = /Applications/MATLAB_R2016a.app/bin/matlab

[julia]
extension = .jl
version = 0.5.2
executable = /usr/local/bin/julia

[python]
extension = .py
executable = /Users/uweschmitt/Projects/eawag_datapool/venv/bin/python3

[logging]
config_file = ./logging_config.yaml

[log_receiver]
port = 5559

[worker]
port = 5555
count = 5

[server]
pid_file = /Users/uweschmitt/.datapool/datapool.pid

[landing_zone]
folder = /Users/uweschmitt/Projects/eawag_datapool/src/integration_tests/lz

