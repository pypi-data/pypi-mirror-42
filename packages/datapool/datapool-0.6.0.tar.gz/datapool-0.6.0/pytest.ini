[pytest]
addopts = -rs --full-trace
testpaths = tests
