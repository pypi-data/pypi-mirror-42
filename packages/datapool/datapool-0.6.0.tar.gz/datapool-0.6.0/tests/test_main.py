# encoding: utf-8
from __future__ import absolute_import, division, print_function

import os
import re
import shutil
import subprocess
import time
from fnmatch import fnmatch

import pytest
from click.testing import CliRunner

from datapool.utils import find_executable

has_matlab = find_executable("matlab") is not None


def test_init_config(data_path, tmpdir, monkeypatch, regtest_list_folders, regtest):

    regtest.identifier = "with_matlab" if has_matlab else "without_matlab"

    monkeypatch.setenv("ETC", tmpdir.strpath)
    from datapool.main import init_config

    runner = CliRunner()

    fresh_landing_zone = tmpdir.join("landing_zone").mkdir().strpath
    result = runner.invoke(init_config, [fresh_landing_zone])
    assert result.exit_code == 0, result.output

    regtest.write(result.output)

    regtest_list_folders(fresh_landing_zone, recurse=True)

    # try the same folder again:
    result = runner.invoke(init_config, [fresh_landing_zone])
    output = result.output.strip().split("\n")

    assert fnmatch(output[3], "*folder * already exists"), output
    assert result.exit_code == 1

    # try new folder, should fail because the datapool config folder was created before:
    non_existing_folder = tmpdir.join("landing_zone_0").mkdir().strpath
    result = runner.invoke(init_config, [non_existing_folder])

    output = result.output.strip()
    assert fnmatch(output, "*datapool folder * already exists"), output

    assert result.exit_code == 1


def test_init_db(data_path):

    from datapool.main import init_db

    runner = CliRunner()
    result = runner.invoke(init_db, [])
    assert result.exit_code == 0, result.output + str(result.exception)

    # should fail: can not init same db again
    runner = CliRunner()
    result = runner.invoke(init_db, [])
    assert result.exit_code == 1

    # should fail, one "--force" is not enough
    result = runner.invoke(init_db, ["--force"])
    assert result.exit_code == 1

    # should succed: we agree to overwrite existing db
    result = runner.invoke(init_db, ["--force", "--force"])
    assert result.exit_code == 0


def test_start_develop(data_path, tmpdir, monkeypatch):
    monkeypatch.setenv("ETC", tmpdir.strpath)

    from datapool.main import start_develop, init_config

    runner = CliRunner()

    fresh_landing_zone = tmpdir.join("landing_zone").mkdir().strpath
    result = runner.invoke(init_config, [fresh_landing_zone])
    assert result.exit_code == 0, result.output

    runner = CliRunner()
    result = runner.invoke(start_develop, [tmpdir.join("test_start_develop").strpath])
    assert result.exit_code == 0, result.output + str(result.exception)


def test_update_operational(data_path, tmpdir, monkeypatch):

    from datapool.main import update_operational, start_develop, init_config

    monkeypatch.setenv("ETC", tmpdir.strpath)

    runner = CliRunner()

    fresh_landing_zone = tmpdir.join("landing_zone").mkdir().strpath
    result = runner.invoke(init_config, [fresh_landing_zone])
    assert result.exit_code == 0, result.output

    result = runner.invoke(start_develop, [tmpdir.join("test_start_develop").strpath])
    assert result.exit_code == 0, result.output + str(result.exception)

    # no assert for return code. might fail or not depending on matlab availability
    runner.invoke(update_operational, [tmpdir.join("test_start_develop").strpath])


def test_check(data_path, tmpdir, monkeypatch):

    from datapool.main import check, start_develop, init_config

    monkeypatch.setenv("ETC", tmpdir.strpath)

    runner = CliRunner()

    fresh_landing_zone = tmpdir.join("landing_zone").mkdir().strpath
    result = runner.invoke(init_config, [fresh_landing_zone])
    assert result.exit_code == 0, result.output

    result = runner.invoke(start_develop, [tmpdir.join("test_start_develop").strpath])
    assert result.exit_code == 0, result.output + str(result.exception)

    # no assert for return code. might fail or not depending on matlab availability
    runner.invoke(check, [tmpdir.join("test_start_develop").strpath])


@pytest.mark.skipif(not has_matlab, reason="needs matlab to work")
def test_all(regtest_print, tmpdir, monkeypatch):

    import signal

    from datapool.main import (
        init_config,
        init_db,
        create_example,
        update_operational,
        delete_entity,
        check_config,
        check,
        delete_signals,
    )

    monkeypatch.setenv("ETC", tmpdir.strpath)

    runner = CliRunner()

    fresh_landing_zone = tmpdir.join("landing_zone").mkdir().strpath
    dlz = tmpdir.join("dlz")

    def invoke(cmd, args=(), assert_fail=False, sleep=0):
        time.sleep(sleep)
        result = runner.invoke(cmd, args, catch_exceptions=False)
        print(result.output)
        regtest_print(result.output.rstrip())
        if assert_fail:
            assert result.exit_code != 0, result.output
        else:
            assert result.exit_code == 0, result.output

    invoke(init_config, [fresh_landing_zone, "--use-sqlitedb"])
    invoke(init_db)
    invoke(check_config)
    invoke(create_example, [dlz.strpath])
    invoke(check, [dlz.strpath])

    def start_bg_server():

        proc = subprocess.Popen(
            ["pool", "run-simple-server"],
            stdout=subprocess.PIPE,
            stderr=subprocess.STDOUT,
        )

        for line in iter(proc.stdout.readline, b""):
            print("SERVER", line.rstrip())
            if line.startswith(b"- observe"):
                # server startup finihsed
                break
        return proc

    def wait_for_conversion_finishes(proc):
        for line in iter(proc.stdout.readline, b""):
            print("SERVER", line.rstrip())
            line = str(line, "utf-8").rstrip()
            if re.match("committed \d+ signals", line.strip()):
                break

    proc = start_bg_server()

    try:
        invoke(update_operational, [dlz.strpath])
        shutil.copy(
            dlz.join(
                "data/sensor_from_company_xyz/sensor_instance_matlab/raw_data/"
                "data-001.raw"
            ).strpath,
            tmpdir.join(
                "landing_zone/data/sensor_from_company_xyz/sensor_instance_matlab/"
                "raw_data"
            ).strpath,
        )
        wait_for_conversion_finishes(proc)
        time.sleep(.5)
        # patch getuid, to make delete command work
        os.getuid = lambda: 0
        invoke(delete_signals, ["timestamp>=2000-01-01:00:00:00,parameter==Flow"])
        invoke(
            delete_signals,
            ["--force", "--force", "timestamp>=2000-01-01:00:00:00,parameter==Flow"],
        )
        invoke(delete_entity, ["--what", "site", "test_site"])
        invoke(delete_entity, ["--force", "--force", "--what", "site", "test_site"])
        invoke(update_operational, [dlz.strpath], assert_fail=True)

        shutil.copy(
            dlz.join(
                "data/sensor_from_company_xyz/sensor_instance_matlab/"
                "raw_data/data-001.raw"
            ).strpath,
            tmpdir.join(
                "landing_zone/data/sensor_from_company_xyz/sensor_instance_matlab/"
                "raw_data/data-002.raw"
            ).strpath,
        )

        for line in iter(proc.stdout.readline, b""):
            print("SERVER", line)
            line = str(line, "utf-8").rstrip()
            if "dispatch" in line and "raw_data/data-002.raw" in line:
                time.sleep(.5)
                break

    finally:
        os.kill(proc.pid, signal.SIGTERM)
        lines = []
        for line in iter(proc.stdout.readline, b""):
            line = str(line, "utf-8").rstrip()
            lines.append(line)
        assert any("waiting for dispatcher to finish" in l for l in lines)
        assert any("dispatch done" in l for l in lines)
        assert any("shutdown done" in l for l in lines)
