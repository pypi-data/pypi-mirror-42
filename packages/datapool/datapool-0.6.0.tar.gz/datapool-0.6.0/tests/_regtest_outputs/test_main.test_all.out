
> this is datapool version X.Y.Z
> init-config
- guess settings
- created config files at <tmpdir_from_fixture>/datapool
  please edit these files and adapt the data base configuration to your setup

> this is datapool version X.Y.Z
> init-db
+ intialized db

> this is datapool version X.Y.Z
> check-config
- check settings in config file <tmpdir_from_fixture>/datapool/datapool.ini
- check landing zone '<tmpdir_from_fixture>/landing_zone'
- try to connect to db
  - connected to db
- check R configuration + code execution
- R interpreter works
- check python configuration + code execution
- python interpreter works
- check matlab configuration + code execution
- matlab interpreter works
- check julia configuration + code execution
- julia interpreter works
- checked julia version.
+ all checks passed

> this is datapool version X.Y.Z
> create-example
- setup example landing zone
+ example landing zone created

> this is datapool version X.Y.Z
> check
- check names and places of changed files at landing zone <tmpdir_from_fixture>/dlz
- file names and their places are ok
- check yaml files in landing zone at <tmpdir_from_fixture>/dlz
- copy meta data from productive db
  - copy table parameter
  - copy table parameter_averaging
  - copy table picture
  - copy table site
  - copy table source
  - copy table source_type
  - copy table special_value_definition
- check folder structure
  - sites is ok
  - data is ok
  - data/sensor_from_company_xyz/sensor_instance_python/raw_data is ok
  - data/sensor_from_company_xyz/sensor_instance_r/raw_data is ok
  - data/sensor_from_company_xyz/sensor_instance_matlab/raw_data is ok
  - data/sensor_from_company_xyz/sensor_instance_julia/raw_data is ok
- detected 7 yaml files:
  - <tmpdir_from_fixture>/dlz/data/parameters.yaml
  - <tmpdir_from_fixture>/dlz/data/sensor_from_company_xyz/source_type.yaml
  - <tmpdir_from_fixture>/dlz/data/sensor_from_company_xyz/sensor_instance_julia/source.yaml
  - <tmpdir_from_fixture>/dlz/data/sensor_from_company_xyz/sensor_instance_matlab/source.yaml
  - <tmpdir_from_fixture>/dlz/data/sensor_from_company_xyz/sensor_instance_python/source.yaml
  - <tmpdir_from_fixture>/dlz/data/sensor_from_company_xyz/sensor_instance_r/source.yaml
  - <tmpdir_from_fixture>/dlz/sites/example_site/site.yaml
- check yaml files
- all yaml files checked
- check scripts landing zone at <tmpdir_from_fixture>/dlz
- check data/sensor_from_company_xyz/sensor_instance_julia/conversion.jl on data/sensor_from_company_xyz/sensor_instance_julia/raw_data/data-001.raw
  - wrote conversion result as csv to <tmpdir_from_tempfile_module>/sensor_instance_julia_0.csv
  - wrote conversion result as txt to <tmpdir_from_tempfile_module>/sensor_instance_julia_0.txt
  - first conversion needed <TIME> msec
  - second conversion needed <TIME> msec
- check data/sensor_from_company_xyz/sensor_instance_matlab/conversion.m on data/sensor_from_company_xyz/sensor_instance_matlab/raw_data/data-001.raw
  - wrote conversion result as csv to <tmpdir_from_tempfile_module>/sensor_instance_matlab_0.csv
  - wrote conversion result as txt to <tmpdir_from_tempfile_module>/sensor_instance_matlab_0.txt
  - first conversion needed <TIME> msec
  - second conversion needed <TIME> msec
- check data/sensor_from_company_xyz/sensor_instance_python/conversion.py on data/sensor_from_company_xyz/sensor_instance_python/raw_data/data-001.raw
  - wrote conversion result as csv to <tmpdir_from_tempfile_module>/sensor_instance_python_0.csv
  - wrote conversion result as txt to <tmpdir_from_tempfile_module>/sensor_instance_python_0.txt
  - first conversion needed <TIME> msec
  - second conversion needed <TIME> msec
- check data/sensor_from_company_xyz/sensor_instance_r/conversion.r on data/sensor_from_company_xyz/sensor_instance_r/raw_data/data-001.raw
  - wrote conversion result as csv to <tmpdir_from_tempfile_module>/sensor_instance_r_0.csv
  - wrote conversion result as txt to <tmpdir_from_tempfile_module>/sensor_instance_r_0.txt
  - first conversion needed <TIME> msec
  - second conversion needed <TIME> msec
- check signals integrity
- check signals against db
- all scripts checked
+ congratulations: all checks succeeded.

> this is datapool version X.Y.Z
> update-operational
- datapool server is running.
- development landing zone seems to be sane.
- check names and places of changed files at landing zone <tmpdir_from_fixture>/dlz
- file names and their places are ok
- check yaml files in landing zone at <tmpdir_from_fixture>/dlz
- copy meta data from productive db
  - copy table parameter
  - copy table parameter_averaging
  - copy table picture
  - copy table site
  - copy table source
  - copy table source_type
  - copy table special_value_definition
- check folder structure
  - sites is ok
  - data is ok
  - data/sensor_from_company_xyz/sensor_instance_python/raw_data is ok
  - data/sensor_from_company_xyz/sensor_instance_r/raw_data is ok
  - data/sensor_from_company_xyz/sensor_instance_matlab/raw_data is ok
  - data/sensor_from_company_xyz/sensor_instance_julia/raw_data is ok
- detected 7 yaml files:
  - <tmpdir_from_fixture>/dlz/data/parameters.yaml
  - <tmpdir_from_fixture>/dlz/data/sensor_from_company_xyz/source_type.yaml
  - <tmpdir_from_fixture>/dlz/data/sensor_from_company_xyz/sensor_instance_julia/source.yaml
  - <tmpdir_from_fixture>/dlz/data/sensor_from_company_xyz/sensor_instance_matlab/source.yaml
  - <tmpdir_from_fixture>/dlz/data/sensor_from_company_xyz/sensor_instance_python/source.yaml
  - <tmpdir_from_fixture>/dlz/data/sensor_from_company_xyz/sensor_instance_r/source.yaml
  - <tmpdir_from_fixture>/dlz/sites/example_site/site.yaml
- check yaml files
- all yaml files checked
- check scripts landing zone at <tmpdir_from_fixture>/dlz
- check data/sensor_from_company_xyz/sensor_instance_julia/conversion.jl on data/sensor_from_company_xyz/sensor_instance_julia/raw_data/data-001.raw
  - wrote conversion result as csv to <tmpdir_from_tempfile_module>/sensor_instance_julia_0.csv
  - wrote conversion result as txt to <tmpdir_from_tempfile_module>/sensor_instance_julia_0.txt
  - first conversion needed <TIME> msec
- check data/sensor_from_company_xyz/sensor_instance_matlab/conversion.m on data/sensor_from_company_xyz/sensor_instance_matlab/raw_data/data-001.raw
  - wrote conversion result as csv to <tmpdir_from_tempfile_module>/sensor_instance_matlab_0.csv
  - wrote conversion result as txt to <tmpdir_from_tempfile_module>/sensor_instance_matlab_0.txt
  - first conversion needed <TIME> msec
- check data/sensor_from_company_xyz/sensor_instance_python/conversion.py on data/sensor_from_company_xyz/sensor_instance_python/raw_data/data-001.raw
  - wrote conversion result as csv to <tmpdir_from_tempfile_module>/sensor_instance_python_0.csv
  - wrote conversion result as txt to <tmpdir_from_tempfile_module>/sensor_instance_python_0.txt
  - first conversion needed <TIME> msec
- check data/sensor_from_company_xyz/sensor_instance_r/conversion.r on data/sensor_from_company_xyz/sensor_instance_r/raw_data/data-001.raw
  - wrote conversion result as csv to <tmpdir_from_tempfile_module>/sensor_instance_r_0.csv
  - wrote conversion result as txt to <tmpdir_from_tempfile_module>/sensor_instance_r_0.txt
  - first conversion needed <TIME> msec
- check signals integrity
- check signals against db
- all scripts checked
+ congratulations: all checks succeeded.
- created folder sites
- created folder sites/example_site
- created folder sites/example_site/images
- copied file sites/example_site/images/24G35_regenwetter.jpg
- copied file sites/example_site/images/IMG_0312.JPG
- copied file sites/example_site/images/IMG_0732.JPG
- copied file sites/example_site/site.yaml
- copied file data/parameters.yaml
- copied file data/sensor_from_company_xyz/source_type.yaml
- copied file data/sensor_from_company_xyz/sensor_instance_julia/source.yaml
- copied file data/sensor_from_company_xyz/sensor_instance_matlab/source.yaml
- copied file data/sensor_from_company_xyz/sensor_instance_python/source.yaml
- copied file data/sensor_from_company_xyz/sensor_instance_r/source.yaml
- copied file data/sensor_from_company_xyz/sensor_instance_julia/conversion.jl
- created folder data/sensor_from_company_xyz/sensor_instance_julia/raw_data
- copied file data/sensor_from_company_xyz/sensor_instance_matlab/conversion.m
- created folder data/sensor_from_company_xyz/sensor_instance_matlab/raw_data
- copied file data/sensor_from_company_xyz/sensor_instance_python/conversion.py
- created folder data/sensor_from_company_xyz/sensor_instance_python/raw_data
- copied file data/sensor_from_company_xyz/sensor_instance_r/conversion.r
- created folder data/sensor_from_company_xyz/sensor_instance_r/raw_data
+ copied/created 21 files/folders to/in <tmpdir_from_fixture>/landing_zone

> this is datapool version X.Y.Z
> delete-signals
- deletion affects 1451 signal(s)

  id    timestamp            value  parameter  source                  site  x       y       z  
  ----  -------------------  -----  ---------  ----------------------  ----  ------  ------  -  
  2903  2013-11-13 10:06:00  39.21  Flow       sensor_instance_matlab        682558  239404     
  2904  2013-11-13 10:08:00  37.6   Flow       sensor_instance_matlab        682558  239404     
  2905  2013-11-13 10:10:00  35.09  Flow       sensor_instance_matlab        682558  239404     
  2906  2013-11-13 10:12:00  36.79  Flow       sensor_instance_matlab        682558  239404     
  2907  2013-11-13 10:14:00  40.63  Flow       sensor_instance_matlab        682558  239404     
  2908  2013-11-13 10:16:00  40.85  Flow       sensor_instance_matlab        682558  239404     
  2909  2013-11-13 10:18:00  37.71  Flow       sensor_instance_matlab        682558  239404     
  2910  2013-11-13 10:20:00  37.3   Flow       sensor_instance_matlab        682558  239404     
  2911  2013-11-13 10:22:00  34.94  Flow       sensor_instance_matlab        682558  239404     
  2912  2013-11-13 10:24:00  33.04  Flow       sensor_instance_matlab        682558  239404     
  
  ... skipped 1441 signals

- use '--force --force' to delete signals.

> this is datapool version X.Y.Z
> delete-signals
- deletion affects 1451 signal(s)

  id    timestamp            value  parameter  source                  site  x       y       z  
  ----  -------------------  -----  ---------  ----------------------  ----  ------  ------  -  
  2903  2013-11-13 10:06:00  39.21  Flow       sensor_instance_matlab        682558  239404     
  2904  2013-11-13 10:08:00  37.6   Flow       sensor_instance_matlab        682558  239404     
  2905  2013-11-13 10:10:00  35.09  Flow       sensor_instance_matlab        682558  239404     
  2906  2013-11-13 10:12:00  36.79  Flow       sensor_instance_matlab        682558  239404     
  2907  2013-11-13 10:14:00  40.63  Flow       sensor_instance_matlab        682558  239404     
  2908  2013-11-13 10:16:00  40.85  Flow       sensor_instance_matlab        682558  239404     
  2909  2013-11-13 10:18:00  37.71  Flow       sensor_instance_matlab        682558  239404     
  2910  2013-11-13 10:20:00  37.3   Flow       sensor_instance_matlab        682558  239404     
  2911  2013-11-13 10:22:00  34.94  Flow       sensor_instance_matlab        682558  239404     
  2912  2013-11-13 10:24:00  33.04  Flow       sensor_instance_matlab        682558  239404     
  
  ... skipped 1441 signals

- deleted 1451 rows

> this is datapool version X.Y.Z
> delete-entity
- following site found in database:
  SiteDbo test_site:
      street      : 'Teststrasse 1'
      city        : '8048 Zurich, Switzerland'
      x, y, z     : '580000', '255300', '455.3'
      description : All the details of the site.All the details the site.All the details of the site.All the details of the site.All the details of the site.All the details of the site.All the details of the site.
      PictureDbo:
          filename    : 'IMG_0312.JPG'
          timestamp   : datetime.datetime(2016, 2, 1, 12, 0)
          filesize    : 356
          description : 'outside'
      PictureDbo:
          filename    : 'IMG_0732.JPG'
          timestamp   : None
          filesize    : 356
          description : 'manholeD1'

- deletion affects 0 signal(s)
- the following files and folders will be deleted in this order:
  <tmpdir_from_fixture>/landing_zone/sites/example_site/images/24G35_regenwetter.jpg
  <tmpdir_from_fixture>/landing_zone/sites/example_site/images/IMG_0312.JPG
  <tmpdir_from_fixture>/landing_zone/sites/example_site/images/IMG_0732.JPG
  <tmpdir_from_fixture>/landing_zone/sites/example_site/site.yaml
  <tmpdir_from_fixture>/landing_zone/sites/example_site/images/
  <tmpdir_from_fixture>/landing_zone/sites/example_site/
- this operation deleted 0 signals
- use '--force --force' to delete test_site

> this is datapool version X.Y.Z
> delete-entity
- following site found in database:
  SiteDbo test_site:
      street      : 'Teststrasse 1'
      city        : '8048 Zurich, Switzerland'
      x, y, z     : '580000', '255300', '455.3'
      description : All the details of the site.All the details the site.All the details of the site.All the details of the site.All the details of the site.All the details of the site.All the details of the site.
      PictureDbo:
          filename    : 'IMG_0312.JPG'
          timestamp   : datetime.datetime(2016, 2, 1, 12, 0)
          filesize    : 356
          description : 'outside'
      PictureDbo:
          filename    : 'IMG_0732.JPG'
          timestamp   : None
          filesize    : 356
          description : 'manholeD1'

- deletion affects 0 signal(s)
- the following files and folders will be deleted in this order:
  <tmpdir_from_fixture>/landing_zone/sites/example_site/images/24G35_regenwetter.jpg
  <tmpdir_from_fixture>/landing_zone/sites/example_site/images/IMG_0312.JPG
  <tmpdir_from_fixture>/landing_zone/sites/example_site/images/IMG_0732.JPG
  <tmpdir_from_fixture>/landing_zone/sites/example_site/site.yaml
  <tmpdir_from_fixture>/landing_zone/sites/example_site/images/
  <tmpdir_from_fixture>/landing_zone/sites/example_site/
- deleted 0 rows from table 'signal'
- removed <tmpdir_from_fixture>/landing_zone/sites/example_site/images/24G35_regenwetter.jpg
- removed <tmpdir_from_fixture>/landing_zone/sites/example_site/images/IMG_0312.JPG
- removed <tmpdir_from_fixture>/landing_zone/sites/example_site/images/IMG_0732.JPG
- removed <tmpdir_from_fixture>/landing_zone/sites/example_site/site.yaml
- removed <tmpdir_from_fixture>/landing_zone/sites/example_site/images
- removed <tmpdir_from_fixture>/landing_zone/sites/example_site
- this operation deleted 0 signals

> this is datapool version X.Y.Z
> update-operational
- datapool server is running.
+ development landing zone invalid: <tmpdir_from_fixture>/dlz/.start_state contains files which are not in <tmpdir_from_fixture>/landing_zone
